eZ/Publish/Core/Persistence/Legacy/Content/UrlAlias/Handler.php

```diff
@@ -1004,4 +1013,61 @@ protected function getHash($text)
     {
         return md5(mb_strtolower($text, 'UTF-8'));
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function archiveUrlAliasesForDeletedTranslations($locationId, $parentLocationId, array $languageCodes)
+    {
+        $parentId = $this->getRealAliasId($parentLocationId);
+
+        // filter removed Translations
+        $urlAliases = $this->listURLAliasesForLocation($locationId);
+        $removedLanguages = [];
+        foreach ($urlAliases as $urlAlias) {
+            foreach ($urlAlias->languageCodes as $languageCode) {
+                if (!in_array($languageCode, $languageCodes)) {
+                    $removedLanguages[] = $languageCode;
+                }
+            }
+        }
+
+        if (empty($removedLanguages)) {
+            return;
+        }
+
+        // map languageCodes to their IDs
+        $languageIds = array_map(
+            function ($languageCode) {
+                return $this->languageHandler->loadByLanguageCode($languageCode)->id;
```

Comment:
todo, we need to have a multi load spi for this to get rid of these calls, but this can be marked for followup, I might be able to handle it end of september

Reply:
followup is fine